monkey/tests/unit_tests/infection_monkey/puppet/test_puppet.py

```diff
@@ -17,10 +25,18 @@ def mock_plugin_registry() -> PluginRegistry:
     )
 
 
+@pytest.fixture
+def plugin_compatability_verifier() -> PluginCompatabilityVerifier:
+    return PluginCompatabilityVerifier(
+        MagicMock(spec=IIslandAPIClient), deepcopy(FAKE_HARD_CODED_PLUGIN_MANIFESTS)
+    )
```

Comment:
Instead of returning a real `PluginCompatabilityVerifier` with mocked out parameters, just return a mock. Then you can manipulate the mock directly instead of manipulating the components that make it up.